core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -147,15 +151,16 @@ void existing(ManifestEntry<F> entry) {
    * <p>The entry's snapshot ID will be this manifest's snapshot ID.
    *
    * @param deletedFile a file
+   * @param dataSequenceNumber a data sequence number for the file
    */
-  public void delete(F deletedFile) {
-    addEntry(reused.wrapDelete(snapshotId, deletedFile));
+  public void delete(F deletedFile, long dataSequenceNumber) {
```

Comment:
It breaks the API in core but I don't think we can default it to anything.

Reply:
Can we keep this method, but throw `UnsupportedOperationException`? That way it isn't a compile or linker error, we can send a reasonable message explaining why.

Reply:
Yeah, the other option is passing `null`, but that would effectively set the data sequence number to `0` for the deleted file, which is not correct for reconstructing when the data was added. Better to throw an exception.

Reply:
Let's throw an exception.

Reply:
Done.